Home screen with beehives backend

4chiAA opened the pull request.

4chiAA committed:
add model beehive and enum beehive type

4chiAA committed:
add repo, service and controller

4chiAA committed:
add method getAllBeehives in service and matching GetMapping in controller

4chiAA committed:
add flapdoodle and integration test for @GetMapping getAllbeehives

4chiAA committed:
removed unnecessary imports

4chiAA committed:
added BeehiveServiceTest with one UnitTest

4chiAA committed:
add second UnitTest

lostspezi commented on backend/pom.xml:
```diff
@@ -27,6 +27,12 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-web</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>de.flapdoodle.embed</groupId>
+			<artifactId>de.flapdoodle.embed.mongo.spring31x</artifactId>
+			<version>4.10.1</version>
+			<scope>Test</scope>
+		</dependency>
```
There is a newer Version, check out Slack, Sara posted the correct Version :)

lostspezi commented on backend/src/main/java/com/beenabler/backend/controller/BeehiveController.java:
```diff
@@ -0,0 +1,23 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
```
```suggestion
@RequestMapping("/api/beehives")
```

lostspezi commented on backend/src/main/java/com/beenabler/backend/controller/BeehiveController.java:
```diff
@@ -0,0 +1,23 @@
+
+    private final BeehiveService beehiveService;
+
+    @GetMapping("/beehives")
```
```suggestion
    @GetMapping
```

lostspezi commented on backend/src/main/java/com/beenabler/backend/model/Beehive.java:
```diff
@@ -0,0 +1,17 @@
+    private String name;
+    private String location;
+    private BeehiveType type;
+}
```
No new line at end of file

lostspezi commented on backend/src/main/java/com/beenabler/backend/model/BeehiveType.java:
```diff
@@ -0,0 +1,12 @@
+    private BeehiveType(String type){
+        this.type = type;
+    }
+}
```
No new line at end of file

lostspezi commented on backend/src/main/java/com/beenabler/backend/repo/BeehiveRepo.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
You don´t need the annotation, because it is already included in the MongoRepository
```suggestion
```

lostspezi commented on backend/src/test/java/com/beenabler/backend/controller/BeehiveControllerTest.java:
```diff
@@ -0,0 +1,29 @@
+    @DirtiesContext
+    void getAllBeehives_whenCalledInitially_thenReturnEmptyList() throws Exception {
+        //WHEN
+        mockMvc.perform(MockMvcRequestBuilders.get(BASE_URL+"/beehives"))
```
```suggestion
        mockMvc.perform(MockMvcRequestBuilders.get(BASE_URL))
```

lostspezi commented on backend/src/test/java/com/beenabler/backend/controller/BeehiveControllerTest.java:
```diff
@@ -0,0 +1,29 @@
+@AutoConfigureMockMvc
+class BeehiveControllerTest {
+
+    private final String BASE_URL = "/api";
```
```suggestion
    private final String BASE_URL = "/api/beehives";
```

4chiAA replied:
> 

Later I want to include GetMapping for single beehives, which would be /api/beehive/{id}

lostspezi commented on backend/src/test/resources/application.properties:
```diff
@@ -0,0 +1 @@
+de.flapdoodle.mongodb.embedded.version=7.0.0
```
No new line at end of file

lostspezi reviewed (changes requested):
Nice Job so far, some recommendations below! It is historical that at the end of a file there needs to be a new line for some reason while compiling. You can read about it [here](https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file)

4chiAA committed:
Update backend/src/main/java/com/beenabler/backend/controller/BeehiveController.java

Later I was planning to read single beehives, which would correspond to /api/beehive/{id}

Co-authored-by: Marcell <marcell.dechant@protonmail.com>

4chiAA committed:
changed requests

4chiAA committed:
changed requests

lostspezi commented on backend/src/main/java/com/beenabler/backend/controller/BeehiveController.java:
It is not best practice to set the URI like this "/api" only and set the "/behivees" to the GetMapping. You will need to set at every Endpoint than "/behivees" instead of setting it only onetime

4chiAA committed:
update GetMapping path

lostspezi reviewed (approved):
LGTM!:)
